src/chef-server-ctl/lib/chef_server_ctl/config.rb

```diff
@@ -0,0 +1,170 @@
+require 'chef_server_ctl/log'
+
+# ChefServerCtl::Config is a global configuration class for
+# ChefServerCtl subcommands.
+#
+# We use a global at the moment to avoid too much upheaval in the
+# various subcommands.
+#
+# Configuration is based on environment variables to make it easy to
+# implement wrappers in our Habitat packaged versions of chef-server.
+#
+# If the environment variables become too unwieldy, we can change them
+# as long as we remember to go fix the Habitat wrappers.
+#
+# TODO(ssd) 2018-08-08: Maybe use a configuration file instead?  I've
+# opted against it for now to avoid having to write yet-another-file
+# out during reconfiugration.
+module ChefServerCtl
+  module Config
+    DEFAULT_KNIFE_CONFIG_FILE = "/etc/opscode/pivotal.rb".freeze
+    DEFAULT_KNIFE_BIN = "/opt/opscode/embedded/bin/knife".freeze
+    DEFAULT_LB_URL = "https://127.0.0.1".freeze
+    DEFAULT_FIPS_LB_URL = "http://127.0.0.1".freeze
+    DEFAULT_RABBITMQCTL_BIN = "/opt/opscode/embedded/service/rabbitmq/sbin/rabbitmqctl".freeze
+    DEFAULT_ERCHEF_REINDEX_SCRIPT = "/opt/opscode/embedded/service/opscode-erchef/bin/reindex-opc-organization".freeze
+
+    def self.init(ctl)
+      @@ctl = ctl
+      Log.debug("Using KNIFE_CONFIG_FILE=#{self.knife_config_file}")
+      Log.debug("Using KNIFE_BIN=#{self.knife_bin}")
+      Log.debug("Using RABBITMQCTL_BIN=#{self.rabbitmqctl_bin}")
+      Log.debug("Using ERCHEF_REINDEX_SCRIPT=#{self.erchef_reindex_script}")
+      Log.debug("Using HABITAT_MODE=#{self.habitat_mode}")
+      # We don't always get run with a full chef-server-running.json
+      # so any setting that fallsback to running_config or
+      # credentials we can't print here. :(
+      # Log.debug("Using BIFROST_URL=#{self.bifrost_url}")
+      # Log.debug("Using LB_URL=#{self.lb_url}")
+    end
+
+    # knife_config should be the path to a configuration file that
+    # allows the `knife` executable to run with pivotal permissions.
+    def self.knife_config_file
+      if ENV['CSC_KNIFE_CONFIG_FILE']
+        ENV['CSC_KNIFE_CONFIG_FILE']
+      else
+        DEFAULT_KNIFE_CONFIG_FILE
+      end
+    end
+
+    # knife_bin is the command used to execute knife.
+    def self.knife_bin
+      if ENV['CSC_KNIFE_BIN']
+        ENV['CSC_KNIFE_BIN']
+      else
+        DEFAULT_KNIFE_BIN
+      end
+    end
+
+    # rabbitmqctl_bin is the command used to execute rabbitmqctl. This
+    # is used for the --wait flag of the reindex command.
+    def self.rabbitmqctl_bin
+      if ENV['CSC_RABBITMQCTL_BIN']
+        ENV['CSC_RABBITMQCTL_BIN']
+      else
+        DEFAULT_RABBITMQCTL_BIN
+      end
+    end
+
+    # fips_enabled indicates whether the chef-server is running in
+    # fips mode.
+    def self.fips_enabled
```

Comment:
Jenkins omnibus tests fail near here. That build was for the a339ca4 change, but I don't see anything obvious to fix it in later commits. I'll dig into things once I finish debugging the build failures on master.
```
    chef-server: /opt/opscode/embedded/lib/ruby/gems/2.5.0/gems/chef-server-ctl-1.0.0/lib/chef_server_ctl/config.rb:72:in `fips_enabled': undefined method `[]' for nil:NilClass (NoMethodError)
    chef-server: 	from /opt/opscode/embedded/lib/ruby/gems/2.5.0/gems/chef-server-ctl-1.0.0/lib/chef_server_ctl/config.rb:81:in `lb_url'
    chef-server: 	from /opt/opscode/embedded/lib/ruby/gems/2.5.0/gems/chef-server-ctl-1.0.0/lib/chef_server_ctl/config.rb:33:in `init'
    chef-server: 	from /opt/opscode/embedded/lib/ruby/gems/2.5.0/gems/chef-server-ctl-1.0.0/bin/chef-server-ctl:267:in `<top (required)>'
    chef-server: 	from /opt/opscode/embedded/bin/chef-server-ctl:23:in `load'
    chef-server: 	from /opt/opscode/embedded/bin/chef-server-ctl:23:in `<main>'
```

Reply:
Look like running_config isn't set up at this point in the code (we've just created a new Omnibus::ChefServerCtl in line 264 of /opt/opscode/embedded/lib/ruby/gems/2.5.0/gems/chef-server-ctl-1.0.0/bin/chef-server-ctl

Reply:
@markan this should be fixed since we no longer try to print lb_url in init

Reply:
Confirmed that the tests are now passing for at least x86: http://wilson.ci.chef.co/job/chef-server-test/253/